packages/@webex/plugin-meetings/src/reachability/index.ts

```diff
@@ -395,16 +409,207 @@ export default class Reachability {
+      },
+    };
+    Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.REACHABILITY_COMPLETED, stats);
+  }
+
+  /**
+   * Starts all the timers used for various timeouts
+   *
+   * @returns {void}
+   */
+  private startTimers() {
+    this.vmnTimer = setTimeout(() => {
+      this.vmnTimer = undefined;
+      // if we are only missing VMN results, then we don't want to wait for them any longer
+      // as they are likely to fail if users are not on corporate network
+      if (this.areAllPublicClusterResultsReady()) {
+        LoggerProxy.logger.log(
+          'Reachability:index#startTimers --> Reachability checks timed out (VMN timeout)'
+        );
+
+        this.resolveReachabilityPromise();
+      }
+    }, VIDEO_MESH_TIMEOUT * 1000);
+
+    this.publicCloudTimer = setTimeout(() => {
+      this.publicCloudTimer = undefined;
+
+      LoggerProxy.logger.log(
+        `Reachability:index#startTimers --> Reachability checks timed out (${DEFAULT_TIMEOUT}s)`
+      );
+
+      // resolve the promise, so that the client won't be blocked waiting on meetings.register() for too long
+      this.resolveReachabilityPromise();
+    }, DEFAULT_TIMEOUT * 1000);
+
+    this.overallTimer = setTimeout(() => {
+      this.overallTimer = undefined;
+      this.abortClusterReachability();
+      this.emit(
+        {
+          file: 'reachability',
+          function: 'overallTimer timeout',
+        },
+        'reachability:done',
+        {}
+      );
+      this.sendMetric();
+
+      LoggerProxy.logger.log(
+        `Reachability:index#startTimers --> Reachability checks fully timed out (${OVERALL_TIMEOUT}s)`
+      );
+    }, OVERALL_TIMEOUT * 1000);
+  }
+
+  /**
+   * Stores given reachability results in local storage
+   *
+   * @param {ReachabilityResults} results
+   * @returns {Promise<void>}
+   */
+  private async storeResults(results: ReachabilityResults) {
+    // @ts-ignore
+    await this.webex.boundedStorage.put(
+      this.namespace,
+      REACHABILITY.localStorageResult,
+      JSON.stringify(results)
+    );
+  }
+
+  /**
+   * Resets all the internal counters that keep track of the results
+   *
+   * @returns {void}
+   */
+  private resetResultCounters() {
+    this.expectedResultsCount.videoMesh.udp = 0;
+    this.expectedResultsCount.public.udp = 0;
+    this.expectedResultsCount.public.tcp = 0;
+    this.expectedResultsCount.public.xtls = 0;
+
+    this.resultsCount.videoMesh.udp = 0;
+    this.resultsCount.public.udp = 0;
+    this.resultsCount.public.tcp = 0;
+    this.resultsCount.public.xtls = 0;
+  }
+
   /**
    * Performs reachability checks for all clusters
    * @param {ClusterList} clusterList
-   * @returns {Promise<ReachabilityResults>} reachability check results
+   * @returns {Promise<void>} promise that's resolved as soon as the checks are started
    */
-  private async performReachabilityChecks(clusterList: ClusterList): Promise<ReachabilityResults> {
```

Comment:
Why did we remove the return type?

Reply:
Previously gatherReachability() was calling performReachabilityChecks() and that was returning all the results that were then stored in localStorage by gatherReachability() in one go. Now we update results in local storage every time a new result comes in through the resultReady event, so gatherReachability() doesn't need to get the results from performReachabilityChecks() anymore and performReachabilityChecks() doesn't wait for all results to be ready before it resolves and only gatherReachability() waits for that. Also performReachabilityChecks() is a private function so we can change what it returns at any time.